src/Arcus.POC.Observability.Telemetry.Serilog.Sinks.ApplicationInsights/Converters/OperationContextConverter.cs

```diff
@@ -21,41 +22,71 @@ public class OperationContextConverter
                 return;
             }
 
-            if (telemetryEntry.Properties.TryGetValue(ContextProperties.Correlation.OperationId, out string operationId))
+            // TODO: custom request telemetry assignment
+            if (telemetryEntry is RequestTelemetry requestTelemetry)
             {
-                telemetryEntry.Context.Operation.Id = operationId;
-            }
+                if (telemetryEntry.Properties.TryGetValue(ContextProperties.Correlation.OperationId, out string operationId))
+                {
+                    requestTelemetry.Id = operationId ?? $"generated-{Guid.NewGuid()}";
+                }
+
+                if (telemetryEntry.Properties.TryGetValue(ContextProperties.Correlation.TransactionId, out string transactionId))
+                {
+                    telemetryEntry.Context.Operation.Id = transactionId;
+                }
 
-            if (telemetryEntry.Properties.TryGetValue(ContextProperties.Correlation.OperationParentId, out string operationParentId))
+                if (telemetryEntry.Properties.TryGetValue(ContextProperties.Correlation.OperationParentId, out string operationParentId))
+                {
+                    telemetryEntry.Context.Operation.ParentId = operationParentId;
+                }
+            }
+            else
             {
-                telemetryEntry.Context.Operation.ParentId = operationParentId;
+                if (telemetryEntry.Properties.TryGetValue(ContextProperties.Correlation.TransactionId, out string transactionId))
+                {
+                    telemetryEntry.Context.Operation.Id = transactionId;
+                }
+
+                if (telemetryEntry.Properties.TryGetValue(ContextProperties.Correlation.OperationId, out string operationId))
+                {
+                    telemetryEntry.Context.Operation.ParentId = operationId;
+                }
             }
+        }
```

Comment:
This is actually the most major change.

Reply:
I have seen the screenshot with the results, so I assume this is ok :)

I also assume that this code will need to be added in the Arcus lib itself ?  I would remove the 'generated-' prefix in the final code then.

Reply:
😊 yeah, I added this `generated-abc123` stuff bc I would know when it was used 😅 , I guess that we can reuse the functionality available already in Arcus for generation such ID's, so we'll be fine. 👌